test/serialize.cpp

```diff
@@ -340,3 +340,49 @@ TEST(serialize, non_zero_node_output)
     EXPECT_EQ(topk_out.get_index(), 1);
     EXPECT_EQ(topk_out.get_node()->description(), "TopK");
 }
+
+TEST(serialize, tensor_iterator)
+{
+    // That which we iterate over
+    auto X = make_shared<op::Parameter>(element::f32, Shape{32, 40, 10});
+
+    // Common to all cells
+    auto WH = make_shared<op::Parameter>(element::f32, Shape{20, 20});
+    auto WX = make_shared<op::Parameter>(element::f32, Shape{10, 20});
+    auto bH = make_shared<op::Parameter>(element::f32, Shape{20});
+    auto WY = make_shared<op::Parameter>(element::f32, Shape{20, 5});
+    auto bY = make_shared<op::Parameter>(element::f32, Shape{5});
+
+    // Initial values
+    auto Hinit = make_shared<op::Parameter>(element::f32, Shape{20});
+
+    // Set up the cell body, a function from (Hi, Xi) -> (Ho, Yo)
+    // Cell parameters
+    auto Hi = make_shared<op::Parameter>(element::f32, Shape{32, 20});
+    auto Xi = make_shared<op::Parameter>(element::f32, Shape{32, 1, 10});
+
+    // Body
+    auto Ho = make_shared<op::Relu>(
+        make_shared<op::Dot>(make_shared<op::Reshape>(Xi, AxisVector{0, 1, 2}, Shape{32, 10}), WX) +
+        make_shared<op::Dot>(Hi, WH) + make_shared<op::Broadcast>(bH, Shape{32, 20}, AxisSet{0}));
+    auto Yo = make_shared<op::Relu>(make_shared<op::Dot>(Ho, WY) +
+                                    make_shared<op::Broadcast>(bY, Shape{32, 5}, AxisSet{0}));
+
+    auto tensor_iterator = make_shared<op::TensorIterator>();
```

Comment:
As far as I know, in order to create any nGraph operation you should specify all inputs. 
Why TwnsorIterator have another creation logic? It feels inconsistent with nGraph philosophy.

Reply:
I have been adding no-arg constructors so that you could create the nodes and then connect them together at some point.
But, in this case, we are trying to separate how TensorIterator is implemented from how it is set up so that we can more easily adjust the implementation when we have more experience with it.